guava/src/com/google/common/cache/LocalCache.java

```diff
@@ -2184,12 +2184,7 @@ V lockedGetOrLoad(K key, int hash, CacheLoader<? super K, V> loader) throws Exec
 
       if (createNewEntry) {
         try {
-          // Synchronizes on the entry to allow failing fast when a recursive load is
-          // detected. This may be circumvented when an entry is copied, but will fail fast most
-          // of the time.
-          synchronized (e) {
-            return loadSync(key, hash, loadingValueReference, loader);
-          }
+          return loadSync(key, hash, loadingValueReference, loader);
```

Comment:
Sorry for this post merge comment, I was expecting to find this change on current Guava release, however it looks like the current version on master is still using `synchronized` (see below). Was this change reverted?   


https://github.com/google/guava/blob/master/guava/src/com/google/common/cache/LocalCache.java#L2186-L2191

Reply:
No problem at all. 

Indeed, we reverted this in https://github.com/google/guava/releases/tag/v33.1.0 because of a bug: https://github.com/google/guava/pull/6851#issuecomment-1931276822

From what I understand, the JDK issue will be fixed in Java 24: https://github.com/openjdk/jdk/pull/21565

But I know that's not much help now :/

Reply:
Thanks for the quick reply! it looks like the suggested path is migrating to Caffeine, I'll look into that. Thank you!

Reply:
You're welcome.

To be clear, Caffeine also uses `synchronized`, so its users also need the JDK fix to eliminate the problem with virtual threads. It just offers plenty of other advantages over Guava :)

Reply:
Indeed, due the the dependency with Java ConcurrentHashMap, however it looks like AsynCache is not affected https://github.com/ben-manes/caffeine/issues/779